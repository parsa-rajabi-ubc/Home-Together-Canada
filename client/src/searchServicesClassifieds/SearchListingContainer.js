/**
 * @Author:     Parsa Rajabi
 * @Created:    2021.2.18
 *
 * @Description: Search Listing Container
 *
 */

import React, {useState, createContext, useEffect} from 'react';
import SearchListingFiltersContainer from "./searchFilter/SearchListingFiltersContainer";
import {useHistory} from "react-router-dom";
import ListingResultsContainer from "./listingResults/ListingResultsContainer";
import {resolveCategoryToListingType} from "../common/utils/listingsUtils";
import * as ListingService from "../services/ListingService";
import {getConcatenatedErrorMessage} from "../registration/registrationUtils";
import Loading from "../common/loading/Loading";
import Confirmation from "../common/listings/Confirmation";
import {MEMBER_SERVICE_CATEGORIES} from "../createListing/constants/serviceListingCategoriesText";

export const listingContext = createContext();

export const PAGE_NAMES = {
    SERVICES: "services",
    CLASSIFIEDS: "classifieds",
}

const MESSAGES = {
    CREATE_SEARCH: "Please use filters to search",
    ERROR: "Oops, there was an error loading search results. Please try again",
}

function SearchListingContainer() {

    let URL_PATH = useHistory().location.pathname;
    const [listingPage, setListingPage] = useState('');
    const [searchFiltersSelected, setSearchFiltersSelected] = useState(false);
    const [loading, setLoading] = useState(false);
    const [error, setError] = useState(false);

    useEffect(() => {
        (URL_PATH === "/" + PAGE_NAMES.SERVICES ? setListingPage(PAGE_NAMES.SERVICES) : setListingPage(PAGE_NAMES.CLASSIFIEDS));
    }, [URL_PATH]);

    const onSearch = (searchFilter) => {
        if (!loading) {
            setLoading(true);
            const searchFilterRequestBody = {
                type: resolveCategoryToListingType(searchFilter.selectedCategory),
                category: searchFilter.selectedCategory,
                ...(searchFilter.selectedCategory === MEMBER_SERVICE_CATEGORIES.MEMBER_HOME)
                    ? {subcategories: []} : {subcategories: searchFilter.selectedSubcategories},
                searchArea: searchFilter.searchArea
            }
            ListingService.searchListings(searchFilterRequestBody)
                .then(res => res.json())
                .then(data => {
                    setSearchFiltersSelected(true);
                    if (data.listing) {
                        setError(false);
                        setLoading(false);
                    } else if (data.err) {
                        alert('Error: ' + data.err);
                        setError(true);
                        setLoading(false);
                    } else if (data.errors) {
                        const errorMessage = getConcatenatedErrorMessage(data.errors);
                        // show list of all errors
                        alert(errorMessage);
                        setError(true);
                        setLoading(false);
                    } else {
                        setError(true);
                        setLoading(false);
                    }
                })
                .catch(err => {
                    alert('Error: ' + err.message);
                    setError(true);
                    setLoading(false);
                });
        }
    }

    function showAppropriateResultsPanel() {
        if (loading) {
            return <Loading isLoading={loading}/>
        } else if (!searchFiltersSelected) {
            return <Confirmation message={MESSAGES.CREATE_SEARCH} displayButton={false}/>
        } else if (error) {
            return <Confirmation message={MESSAGES.ERROR} displayButton={false} errorColor={true}/>
        } else {
            return <ListingResultsContainer/>
        }
    }

    return (
        <listingContext.Provider value={listingPage}>
            <div className={"flex"}>
                <div className={"flex-none w-1/3"}>
                    <SearchListingFiltersContainer onSearch={onSearch}/>
                </div>
                <div className={"flex-1"}>
                    {showAppropriateResultsPanel()}
                </div>
            </div>
        </listingContext.Provider>
    );
}

export default SearchListingContainer;
