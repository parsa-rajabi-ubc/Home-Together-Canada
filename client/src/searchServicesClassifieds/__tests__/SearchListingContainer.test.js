/**
 * @Author:     Parsa Rajabi
 * @Created:    2021.2.18
 *
 * @Description: Search Listing Filter Container Test
 *
 */

import React from 'react';
import renderer from 'react-test-renderer'
import SearchListingContainer from "../SearchListingContainer";

jest.mock("react-tooltip/node_modules/uuid", () => ({
            v4: () => "00000000-0000-0000-0000-000000000000"}
    )
);

jest.mock('react-redux', () => ({
    connect: () => {
        return (component) => {
            return component
        };
    }
}));

describe('SearchListingContainer', () => {
    describe('Container test', () => {
        it('should match snapshot test', () => {
            // given
            const props = {
                listingPage: 'services',
                setServiceListingsSearchResults: jest.fn(),
                setClassifiedListingsSearchResults: jest.fn(),
                setServiceListingsSearchFilters: jest.fn(),
                setClassifiedListingsSearchFilters: jest.fn()
            }

            //when
            const component = renderer.create(<SearchListingContainer {...props}/>);
            const tree = component.toJSON();
            //then
            expect(tree).toMatchSnapshot();
        });
    });
});