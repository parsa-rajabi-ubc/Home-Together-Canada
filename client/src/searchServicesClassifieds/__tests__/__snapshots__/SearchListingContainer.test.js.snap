// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchListingContainer Container test should match snapshot test 1`] = `
<div
  className="flex"
>
  <div
    className="flex-none w-1/3"
  >
    <div
      className="col-start-1 col-end-7 py-5 px-5 my-6 ml-6 bg-white shadow-lg rounded-xl"
    >
      <div
        className="grid grid-cols-2 gap-6"
      >
        <div
          className="col-span-3 sm:col-span-2"
        >
          <label
            className="text-2xl font-medium"
          >
            Search Filtering
          </label>
          <label>
            <input
              className="btn btn-green w-1/4 float-right py-2"
              onClick={[Function]}
              type="submit"
              value="Search"
            />
          </label>
          <section
            className="my-8"
          >
            <section
              className="mb-4"
            >
              <label
                className="label text-lg mb-1"
              >
                Search Area
              </label>
              <div>
                <div
                  className="grid grid-cols-10 gap-x-2"
                >
                  <div
                    className="col-start-1 col-end-4"
                  >
                    <div>
                      <div
                        className=" css-2b097c-container"
                        onKeyDown={[Function]}
                      >
                        <div
                          className=" css-1jumw1e-control"
                          onMouseDown={[Function]}
                          onTouchEnd={[Function]}
                        >
                          <div
                            className=" css-g1d714-ValueContainer"
                          >
                            <div
                              className=" css-1k4wfr7-placeholder"
                            >
                              Province
                            </div>
                            <div
                              className="css-b8ldur-Input"
                            >
                              <div
                                className=""
                                style={
                                  Object {
                                    "display": "inline-block",
                                  }
                                }
                              >
                                <input
                                  aria-autocomplete="list"
                                  autoCapitalize="none"
                                  autoComplete="off"
                                  autoCorrect="off"
                                  disabled={false}
                                  id="react-select-2-input"
                                  onBlur={[Function]}
                                  onChange={[Function]}
                                  onFocus={[Function]}
                                  spellCheck="false"
                                  style={
                                    Object {
                                      "background": 0,
                                      "border": 0,
                                      "boxSizing": "content-box",
                                      "color": "inherit",
                                      "fontSize": "inherit",
                                      "label": "input",
                                      "opacity": 1,
                                      "outline": 0,
                                      "padding": 0,
                                      "width": "1px",
                                    }
                                  }
                                  tabIndex="0"
                                  type="text"
                                  value=""
                                />
                                <div
                                  style={
                                    Object {
                                      "height": 0,
                                      "left": 0,
                                      "overflow": "scroll",
                                      "position": "absolute",
                                      "top": 0,
                                      "visibility": "hidden",
                                      "whiteSpace": "pre",
                                    }
                                  }
                                >
                                  
                                </div>
                              </div>
                            </div>
                          </div>
                          <div
                            className=" css-1hb7zxy-IndicatorsContainer"
                          >
                            <span
                              className=" css-43ykx9-indicatorSeparator"
                            />
                            <div
                              aria-hidden="true"
                              className=" css-1px1elb-indicatorContainer"
                              onMouseDown={[Function]}
                              onTouchEnd={[Function]}
                            >
                              <svg
                                aria-hidden="true"
                                className="css-6q0nyr-Svg"
                                focusable="false"
                                height={20}
                                viewBox="0 0 20 20"
                                width={20}
                              >
                                <path
                                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                                />
                              </svg>
                            </div>
                          </div>
                        </div>
                        <input
                          name="province"
                          type="hidden"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    className="col-start-4 col-end-8"
                  >
                    <div>
                      <div
                        className=" css-2b097c-container"
                        onKeyDown={[Function]}
                      >
                        <div
                          className=" css-1jumw1e-control"
                          onMouseDown={[Function]}
                          onTouchEnd={[Function]}
                        >
                          <div
                            className=" css-g1d714-ValueContainer"
                          >
                            <div
                              className=" css-1k4wfr7-placeholder"
                            >
                              City
                            </div>
                            <div
                              className="css-b8ldur-Input"
                            >
                              <div
                                className=""
                                style={
                                  Object {
                                    "display": "inline-block",
                                  }
                                }
                              >
                                <input
                                  aria-autocomplete="list"
                                  autoCapitalize="none"
                                  autoComplete="off"
                                  autoCorrect="off"
                                  disabled={false}
                                  id="react-select-3-input"
                                  onBlur={[Function]}
                                  onChange={[Function]}
                                  onFocus={[Function]}
                                  spellCheck="false"
                                  style={
                                    Object {
                                      "background": 0,
                                      "border": 0,
                                      "boxSizing": "content-box",
                                      "color": "inherit",
                                      "fontSize": "inherit",
                                      "label": "input",
                                      "opacity": 1,
                                      "outline": 0,
                                      "padding": 0,
                                      "width": "1px",
                                    }
                                  }
                                  tabIndex="0"
                                  type="text"
                                  value=""
                                />
                                <div
                                  style={
                                    Object {
                                      "height": 0,
                                      "left": 0,
                                      "overflow": "scroll",
                                      "position": "absolute",
                                      "top": 0,
                                      "visibility": "hidden",
                                      "whiteSpace": "pre",
                                    }
                                  }
                                >
                                  
                                </div>
                              </div>
                            </div>
                          </div>
                          <div
                            className=" css-1hb7zxy-IndicatorsContainer"
                          >
                            <span
                              className=" css-43ykx9-indicatorSeparator"
                            />
                            <div
                              aria-hidden="true"
                              className=" css-1px1elb-indicatorContainer"
                              onMouseDown={[Function]}
                              onTouchEnd={[Function]}
                            >
                              <svg
                                aria-hidden="true"
                                className="css-6q0nyr-Svg"
                                focusable="false"
                                height={20}
                                viewBox="0 0 20 20"
                                width={20}
                              >
                                <path
                                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                                />
                              </svg>
                            </div>
                          </div>
                        </div>
                        <input
                          name="city"
                          type="hidden"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    className="col-start-8 col-end-11"
                  >
                    <div>
                      <div
                        className=" css-2b097c-container"
                        onKeyDown={[Function]}
                      >
                        <div
                          className=" css-1jumw1e-control"
                          onMouseDown={[Function]}
                          onTouchEnd={[Function]}
                        >
                          <div
                            className=" css-g1d714-ValueContainer"
                          >
                            <div
                              className=" css-1k4wfr7-placeholder"
                            >
                              Radius
                            </div>
                            <div
                              className="css-b8ldur-Input"
                            >
                              <div
                                className=""
                                style={
                                  Object {
                                    "display": "inline-block",
                                  }
                                }
                              >
                                <input
                                  aria-autocomplete="list"
                                  autoCapitalize="none"
                                  autoComplete="off"
                                  autoCorrect="off"
                                  disabled={false}
                                  id="react-select-4-input"
                                  onBlur={[Function]}
                                  onChange={[Function]}
                                  onFocus={[Function]}
                                  spellCheck="false"
                                  style={
                                    Object {
                                      "background": 0,
                                      "border": 0,
                                      "boxSizing": "content-box",
                                      "color": "inherit",
                                      "fontSize": "inherit",
                                      "label": "input",
                                      "opacity": 1,
                                      "outline": 0,
                                      "padding": 0,
                                      "width": "1px",
                                    }
                                  }
                                  tabIndex="0"
                                  type="text"
                                  value=""
                                />
                                <div
                                  style={
                                    Object {
                                      "height": 0,
                                      "left": 0,
                                      "overflow": "scroll",
                                      "position": "absolute",
                                      "top": 0,
                                      "visibility": "hidden",
                                      "whiteSpace": "pre",
                                    }
                                  }
                                >
                                  
                                </div>
                              </div>
                            </div>
                          </div>
                          <div
                            className=" css-1hb7zxy-IndicatorsContainer"
                          >
                            <span
                              className=" css-43ykx9-indicatorSeparator"
                            />
                            <div
                              aria-hidden="true"
                              className=" css-1px1elb-indicatorContainer"
                              onMouseDown={[Function]}
                              onTouchEnd={[Function]}
                            >
                              <svg
                                aria-hidden="true"
                                className="css-6q0nyr-Svg"
                                focusable="false"
                                height={20}
                                viewBox="0 0 20 20"
                                width={20}
                              >
                                <path
                                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                                />
                              </svg>
                            </div>
                          </div>
                        </div>
                        <input
                          name="radius"
                          type="hidden"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </section>
            <label
              className="label text-lg mb-1"
            >
              Select Listing Category
            </label>
            <div>
              <div
                className=" css-2b097c-container"
                onKeyDown={[Function]}
              >
                <div
                  className=" css-1jumw1e-control"
                  onMouseDown={[Function]}
                  onTouchEnd={[Function]}
                >
                  <div
                    className=" css-g1d714-ValueContainer"
                  >
                    <div
                      className=" css-1d6d5xl-placeholder"
                    >
                      Select...
                    </div>
                    <div
                      className="css-b8ldur-Input"
                    >
                      <div
                        className=""
                        style={
                          Object {
                            "display": "inline-block",
                          }
                        }
                      >
                        <input
                          aria-autocomplete="list"
                          autoCapitalize="none"
                          autoComplete="off"
                          autoCorrect="off"
                          disabled={false}
                          id="react-select-5-input"
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          spellCheck="false"
                          style={
                            Object {
                              "background": 0,
                              "border": 0,
                              "boxSizing": "content-box",
                              "color": "inherit",
                              "fontSize": "inherit",
                              "label": "input",
                              "opacity": 1,
                              "outline": 0,
                              "padding": 0,
                              "width": "1px",
                            }
                          }
                          tabIndex="0"
                          type="text"
                          value=""
                        />
                        <div
                          style={
                            Object {
                              "height": 0,
                              "left": 0,
                              "overflow": "scroll",
                              "position": "absolute",
                              "top": 0,
                              "visibility": "hidden",
                              "whiteSpace": "pre",
                            }
                          }
                        >
                          
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    className=" css-1hb7zxy-IndicatorsContainer"
                  >
                    <span
                      className=" css-1okebmr-indicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                      className=" css-tlfecz-indicatorContainer"
                      onMouseDown={[Function]}
                      onTouchEnd={[Function]}
                    >
                      <svg
                        aria-hidden="true"
                        className="css-6q0nyr-Svg"
                        focusable="false"
                        height={20}
                        viewBox="0 0 20 20"
                        width={20}
                      >
                        <path
                          d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </section>
        </div>
      </div>
    </div>
  </div>
  <div
    className="flex-1"
  >
    <div
      className="m-6"
    >
      <div>
        <div>
          <a
            href="/undefined/50"
            onClick={[Function]}
          >
            <section
              className="card-container mt-0"
            >
              <img
                alt=""
                className="float-left w-24 h-auto mx-4 "
                src="HTC_Logo.jpg"
              />
              <div
                className="inline align-middle "
              >
                <label
                  className="font-semibold justify-between"
                >
                   
                  Apartments for 2 available
                   
                </label>
                <section
                  className="inline float-right pr-6 justify-end"
                >
                  <time
                    dateTime={"2021-02-14T22:42:39.000Z"}
                  >
                    Feb 14, 2021
                  </time>
                </section>
              </div>
              <div
                className="mx-6 inline items-center justify-between leading-tight font-light"
              >
                <section
                  className="mb-3"
                >
                   
                  Kelowna Apartment Inc.
                   
                </section>
                <section>
                   
                  2 bed 2 bath - furnished
                   
                </section>
              </div>
            </section>
          </a>
          <a
            href="/undefined/51"
            onClick={[Function]}
          >
            <section
              className="card-container mt-0"
            >
              <img
                alt=""
                className="float-left w-24 h-auto mx-4 "
                src="HTC_Logo.jpg"
              />
              <div
                className="inline align-middle "
              >
                <label
                  className="font-semibold justify-between"
                >
                   
                  Apartments for 3 available
                   
                </label>
                <section
                  className="inline float-right pr-6 justify-end"
                >
                  <time
                    dateTime={"2021-01-16T11:42:39.000Z"}
                  >
                    Jan 16, 2021
                  </time>
                </section>
              </div>
              <div
                className="mx-6 inline items-center justify-between leading-tight font-light"
              >
                <section
                  className="mb-3"
                >
                   
                  Kelowna Apartment Inc.
                   
                </section>
                <section>
                   
                  3 bed 2 bath - furnished
                   
                </section>
              </div>
            </section>
          </a>
          <a
            href="/undefined/52"
            onClick={[Function]}
          >
            <section
              className="card-container mt-0"
            >
              <img
                alt=""
                className="float-left w-24 h-auto mx-4 "
                src="HTC_Logo.jpg"
              />
              <div
                className="inline align-middle "
              >
                <label
                  className="font-semibold justify-between"
                >
                   
                  House with 4 rooms available
                   
                </label>
                <section
                  className="inline float-right pr-6 justify-end"
                >
                  <time
                    dateTime={"2021-02-14T16:42:39.000Z"}
                  >
                    Feb 14, 2021
                  </time>
                </section>
              </div>
              <div
                className="mx-6 inline items-center justify-between leading-tight font-light"
              >
                <section
                  className="mb-3"
                >
                   
                  Kelowna House Rentals
                   
                </section>
                <section>
                   
                  4 bed 3 bath - furnished
                   
                </section>
              </div>
            </section>
          </a>
          <a
            href="/undefined/53"
            onClick={[Function]}
          >
            <section
              className="card-container mt-0"
            >
              <img
                alt=""
                className="float-left w-24 h-auto mx-4 "
                src="HTC_Logo.jpg"
              />
              <div
                className="inline align-middle "
              >
                <label
                  className="font-semibold justify-between"
                >
                   
                  Beautiful House On Lakefront For Rent
                   
                </label>
                <section
                  className="inline float-right pr-6 justify-end"
                >
                  <time
                    dateTime={"2021-02-16T23:42:39.000Z"}
                  >
                    Feb 16, 2021
                  </time>
                </section>
              </div>
              <div
                className="mx-6 inline items-center justify-between leading-tight font-light"
              >
                <section
                  className="mb-3"
                >
                   
                  Lakefront Properties
                   
                </section>
                <section>
                   
                  2 bed 2 bath - furnished
                   
                </section>
              </div>
            </section>
          </a>
          <a
            href="/undefined/54"
            onClick={[Function]}
          >
            <section
              className="card-container mt-0"
            >
              <img
                alt=""
                className="float-left w-24 h-auto mx-4 "
                src="HTC_Logo.jpg"
              />
              <div
                className="inline align-middle "
              >
                <label
                  className="font-semibold justify-between"
                >
                   
                  Vacuum Rentals
                   
                </label>
                <section
                  className="inline float-right pr-6 justify-end"
                >
                  <time
                    dateTime={"2021-01-15T11:42:39.000Z"}
                  >
                    Jan 15, 2021
                  </time>
                </section>
              </div>
              <div
                className="mx-6 inline items-center justify-between leading-tight font-light"
              >
                <section
                  className="mb-3"
                >
                   
                  Kelowna Apartment Inc.
                   
                </section>
                <section>
                   
                  2 bed 2 bath - furnished
                   
                </section>
              </div>
            </section>
          </a>
          <a
            href="/undefined/55"
            onClick={[Function]}
          >
            <section
              className="card-container mt-0"
            >
              <img
                alt=""
                className="float-left w-24 h-auto mx-4 "
                src="HTC_Logo.jpg"
              />
              <div
                className="inline align-middle "
              >
                <label
                  className="font-semibold justify-between"
                >
                   
                  Lawn Management Services
                   
                </label>
                <section
                  className="inline float-right pr-6 justify-end"
                >
                  <time
                    dateTime={"2021-02-15T11:42:39.000Z"}
                  >
                    Feb 15, 2021
                  </time>
                </section>
              </div>
              <div
                className="mx-6 inline items-center justify-between leading-tight font-light"
              >
                <section
                  className="mb-3"
                >
                   
                  Lawns-R-Us
                   
                </section>
                <section>
                   
                  2 bed 2 bath - furnished
                   
                </section>
              </div>
            </section>
          </a>
          <a
            href="/undefined/56"
            onClick={[Function]}
          >
            <section
              className="card-container mt-0"
            >
              <img
                alt=""
                className="float-left w-24 h-auto mx-4 "
                src="HTC_Logo.jpg"
              />
              <div
                className="inline align-middle "
              >
                <label
                  className="font-semibold justify-between"
                >
                   
                  Carpet Cleaning Rentals
                   
                </label>
                <section
                  className="inline float-right pr-6 justify-end"
                >
                  <time
                    dateTime={"2021-02-13T18:42:39.000Z"}
                  >
                    Feb 13, 2021
                  </time>
                </section>
              </div>
              <div
                className="mx-6 inline items-center justify-between leading-tight font-light"
              >
                <section
                  className="mb-3"
                >
                   
                  Carpet Cleaners
                   
                </section>
                <section>
                   
                  We offer services, or tool rentals
                   
                </section>
              </div>
            </section>
          </a>
          <ul
            className="flex m-auto w-2/3"
          >
            <li
              className="next-prev-text disabled"
            >
              <a
                aria-disabled="true"
                aria-label="Previous page"
                className="next-prev-link"
                onClick={[Function]}
                onKeyPress={[Function]}
                rel="prev"
                role="button"
                tabIndex="0"
              >
                Previous
              </a>
            </li>
            <li
              className="selected"
            >
              <a
                aria-current="page"
                aria-label="Page 1 is your current page"
                className="page-numbers current-page"
                onClick={[Function]}
                onKeyPress={[Function]}
                role="button"
                tabIndex="0"
              >
                1
              </a>
            </li>
            <li>
              <a
                aria-current={null}
                aria-label="Page 2"
                className="page-numbers"
                onClick={[Function]}
                onKeyPress={[Function]}
                role="button"
                tabIndex="0"
              >
                2
              </a>
            </li>
            <li>
              <a
                aria-current={null}
                aria-label="Page 3"
                className="page-numbers"
                onClick={[Function]}
                onKeyPress={[Function]}
                role="button"
                tabIndex="0"
              >
                3
              </a>
            </li>
            <li
              className="next-prev-text"
            >
              <a
                aria-disabled="false"
                aria-label="Next page"
                className="next-prev-link"
                onClick={[Function]}
                onKeyPress={[Function]}
                rel="next"
                role="button"
                tabIndex="0"
              >
                Next
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
